.github/workflows/build_and_run_idf_examples.yml

```
@@ -0,0 +1,75 @@
+
+jobs:
+  build:
+    strategy:
+      matrix:
+        # idf_ver: ["release-v5.0", "release-v5.1", "release-v5.2", "release-v5.3", "release-v5.4", "latest"]
+        idf_ver: ["latest"]
+    runs-on: ubuntu-20.04
+    container: espressif/idf:${{ matrix.idf_ver }}
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          submodules: 'true'
+      - name: Build ESP-IDF USB Device examples
+        shell: bash
+        run: |
+          . ${IDF_PATH}/export.sh
+          pip install idf-component-manager==1.5.2 idf-build-apps==2.4.3 --upgrade
+          export EXAMPLES_PATH="${IDF_PATH}/examples/peripherals/usb/device/tusb_hid"
+          export MANIFEST_FILE="${IDF_PATH}/examples/peripherals/.build-test-rules.yml"
+          python .github/ci/override_managed_component.py tinyusb . ${EXAMPLES_PATH}
+          idf-build-apps find --path ${EXAMPLES_PATH}/ --recursive --target esp32s2 --manifest-file ${MANIFEST_FILE} --build-dir build_@t_@w --work-dir @f_@t_@w
+          idf-build-apps build --path ${EXAMPLES_PATH}/ --recursive --target esp32s2 --manifest-file ${MANIFEST_FILE} --build-dir build_@t_@w --work-dir @f_@t_@w
+      - uses: actions/upload-artifact@v4
+        with:
+          name: usb_device_tusb_apps_bin_${{ matrix.idf_ver }}
+          path: |
+              **/*_usb_device_tusb_*/build_esp*/bootloader/bootloader.bin
+              **/*_usb_device_tusb_*/build_esp*/partition_table/partition-table.bin
+              **/*_usb_device_tusb_*/build_esp*/tusb_*.bin
+              **/*_usb_device_tusb_*/build_esp*/build/flasher_args.json
+              **/*_usb_device_tusb_*/build_esp*/config/sdkconfig.json
+          if-no-files-found: error
+  run-target:
+    name: Run pytest for USB Device examples
+    if: ${{ github.repository_owner == 'espressif' }}
+    needs: build
+    strategy:
+      fail-fast: false
+      matrix:
+        # idf_ver: ["release-v5.0", "release-v5.1", "release-v5.2", "release-v5.3", "release-v5.4", "latest"]
+        idf_ver: ["latest"]
+        idf_target: ["esp32s2"]
+        runner_tag: ["usb_device"]
+    runs-on: [self-hosted, linux, docker, "${{ matrix.idf_target }}", "${{ matrix.runner_tag }}"]
+    container:
+      image: espressif/idf:${{ matrix.idf_ver }}
+      options: --privileged --device-cgroup-rule="c 188:* rmw" --device-cgroup-rule="c 166:* rmw" --group-add plugdev
+    steps:
+      - uses: actions/checkout@v4
+      - name: Install Python packages
+        env:
+          PIP_EXTRA_INDEX_URL: "https://dl.espressif.com/pypi/"
+        run: |
+          cd ${IDF_PATH}
+          . ./export.sh
+          pip install --only-binary cryptography pytest-embedded pytest-embedded-serial-esp pytest-embedded-idf pyserial pyusb python-gitlab minio idf-build-apps pytest_ignore_test_results pytest-timeout
+      - uses: actions/download-artifact@v4
+        with:
+          name: usb_device_tusb_apps_bin_${{ matrix.idf_ver }}
+          path: build_${{ matrix.idf_target }}
```

```suggestion
          path: ${{ env.IDF_PATH }}
```

and I think you have to define 

```yaml
env:
  IDF_PATH: /opt/esp/idf
```

in the file, since the `$IDF_PATH` is unexpandable in step's input.